pkg/inventory/inventory-client.go

```diff
@@ -421,3 +431,42 @@ func (cic *ClusterInventoryClient) ApplyInventoryNamespace(obj *unstructured.Uns
 func (cic *ClusterInventoryClient) getMapping(obj *unstructured.Unstructured) (*meta.RESTMapping, error) {
 	return cic.mapper.RESTMapping(obj.GroupVersionKind().GroupKind(), obj.GroupVersionKind().Version)
 }
+
+func (cic *ClusterInventoryClient) updateStatus(obj *unstructured.Unstructured, dryRun common.DryRunStrategy) error {
+	if dryRun.ClientOrServerDryRun() {
+		klog.V(4).Infof("dry-run update inventory status: not updated")
+		return nil
+	}
+	status, found, _ := unstructured.NestedMap(obj.UnstructuredContent(), "status")
+	if !found {
+		return nil
+	}
+
+	klog.V(4).Infof("update inventory status")
+	mapping, err := cic.mapper.RESTMapping(obj.GroupVersionKind().GroupKind())
+	if err != nil {
+		return err
+	}
+	resource := cic.dc.Resource(mapping.Resource).Namespace(obj.GetNamespace())
+	meta := metav1.TypeMeta{
+		Kind:       obj.GetKind(),
+		APIVersion: obj.GetAPIVersion(),
+	}
+	liveObj, err := resource.Get(context.TODO(), obj.GetName(), metav1.GetOptions{TypeMeta: meta}, "status")
+	if err != nil {
+		klog.V(4).Infof("skip updating inventory subresource status for %v: %v", meta, err)
```

Comment:
I think these errors include NotFound.
What error are we expecting exactly if it's not a subresource?

Reply:
This error is the `NOTFOUND` error since the subresource can't be found.

Since the updateStatus is updated after applying the inventory object, we shouldn't get the `NOTFOUND` error because of the inventory object itself doesn't exist.